src/platformdirs/macos.py

```diff
@@ -22,8 +23,21 @@ def user_data_dir(self) -> str:
 
     @property
     def site_data_dir(self) -> str:
-        """:return: data directory shared by users, e.g. ``/Library/Application Support/$appname/$version``"""
-        return self._append_app_name_and_version("/Library/Application Support")
+        """
+        :return: data directory shared by users, e.g. ``/Library/Application Support/$appname/$version``.
+          If we're using a Python binary managed by `Homebrew <https://brew.sh>`_, the directory
+          will be under the Homebrew prefix, e.g. ``/opt/homebrew/share/$appname/$version``.
+          If `multipath <platformdirs.api.PlatformDirsABC.multipath>` is enabled and we're in Homebrew,
+          the response is a multi-path string separated by ":", e.g.
+          ``/opt/homebrew/share/$appname/$version:/Library/Application Support/$appname/$version``
+        """
+        path_list = [self._append_app_name_and_version("/Library/Application Support")]
+        is_homebrew = sys.prefix.startswith("/opt/homebrew")
+        if is_homebrew:
+            path_list.insert(0, self._append_app_name_and_version("/opt/homebrew/share"))
```

Comment:
Rather than inserrt start with homebrew, and use append for the `self._append_app_name_and_version`